app/queries/matched_for_notification.rb

```diff
@@ -7,70 +7,46 @@ def initialize(query_field: :tags, scope: SubscriberList)
   end
 
   # Filter out subscriber lists:
-  # - where the keys from the supplied hash:
-  #   - Have a superset of the keys in the specified query_field for subscriber_lists
-  #   - Have an overlap with  the keys in the specified query_field for or_joined_facet_subscriber_lists
-  # - AND
+  # - where the keys from the supplied hash are a superset of the keys in the specified query_field AND
   # - If the operator is 'any':
   #     the key values in the subscriber list have at least one corresponding match in the supplied hash.
   # - If the operator is 'all'
   #   all key values in the subscriber list have corresponding matches in the supplied hash.
+  # Note that this means that not all keys from the the supplied hash are required to be matched.
   def call(content_item_tags_or_links)
     return [] unless content_item_tags_or_links.present?
 
     content_item_tags_or_links = content_item_tags_or_links.deep_symbolize_keys
 
     only_contains_keys(content_item_tags_or_links.keys).select do |subscriber_list|
-      filter_tags_or_links(subscriber_list, content_item_tags_or_links)
-    end
-  end
-
-private
+      subscriber_list_tags_or_links = subscriber_list.send(@query_field) # send ensures the keys are symbols
 
-  def filter_tags_or_links(subscriber_list, content_item_tags_or_links)
-    operator = subscriber_list.type == 'OrJoinedFacetSubscriberList' ? :any? : :all?
-    subscriber_list_tags_or_links = subscriber_list.send(@query_field) # send ensures the keys are symbols
-    subscriber_list_tags_or_links.keys.send(operator) do |key|
-      content_item_values = Array(content_item_tags_or_links[key])
-      any_values = subscriber_list_tags_or_links[key].fetch(:any, [])
-      all_values = subscriber_list_tags_or_links[key].fetch(:all, [])
+      subscriber_list_tags_or_links.keys.all? do |key|
+        content_item_values = Array(content_item_tags_or_links[key])
+        any_values = subscriber_list_tags_or_links[key].fetch(:any, [])
+        all_values = subscriber_list_tags_or_links[key].fetch(:all, [])
 
-      (all_values - content_item_values).empty? &&
-        (any_values.empty? || (any_values & content_item_values).any?)
+        (all_values - content_item_values).empty? &&
+          (any_values.empty? || (any_values & content_item_values).any?)
+      end
     end
   end
 
-  # Returns:
-  #
-  # all SubscriberLists which have a subset of keys from those requested.
+private
+
+  # Return all SubscriberLists which have a subset of keys from those requested.
   # For example, if `links` is:
   #
   #     {"topics": [...], "organisations": [...]}
   #
   # then this returns all lists which have only "topics" and/or "organisations"
   # links, but not those with "topics" and "policies"
```

Comment:
This comment isn't related to removing "OR" facets. I'd suggest the change in a separate PR.

Reply:
I think this is outside the scope of this PR, since I'm just reverting email-alert-api back to the state it was previously in (which included this comment)

Reply:
Leena beat me to make the same comment by 25 seconds 😁